### PR TITLE
Shared buffer configuration notification

### PR DESCRIPTION
**Pull request by timcnicholls**
This PR handles notification of buffer configuration from frame receiver to processor. Config parameters for the name of the shared memory buffer are also removed from the frame processor.

### CODE REVIEW

**Comment by timcnicholls** [comment]
Closes #37

File: config/fp_py_test.config [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[FrameProcessor]
+ctrl_endpoint=tcp://127.0.0.1:5000
+sharedbuf=FrameReceiverBuffer
```
Comment (ajgdls) [inline config/fp_py_test.config]: Is this sharedbuf entry now redundant?

Comment (ajgdls) [inline config/fp_py_test.config]: Oh actually I misunderstood, this config is not for the FP I guess?

File: frameProcessor/src/SharedMemoryController.cpp [hunk 2]
```diff
@@ -112,29 +150,50 @@ void SharedMemoryController::handleRxChannel()
...
-
-        // Set the frame number
-        frame->set_frame_number(rxMsg.get_param<int>("frame", 0));
-
-        // Loop over registered callbacks, placing the frame onto each queue
-        std::map<std::string, boost::shared_ptr<IFrameCallback> >::iterator cbIter;
-        for (cbIter = callbacks_.begin(); cbIter != callbacks_.end(); ++cbIter) {
-          cbIter->second->getWorkQueue()->add(frame);
+
+        if (sbm_) {
```
Comment (ajgdls) [inline frameProcessor/src/SharedMemoryController.cpp]: This check prevents crash on no valid shared memory registration?  I guess in the real application this is not a normal situation but I should probably write a unit test for this.  Note to self...

**Pull request merged**